identifier.go

```diff
@@ -0,0 +1,62 @@
+package amf0
+
+import "reflect"
+
+// Identifier is a type capable of preforming bi-direcitonal lookups with
+// respect to the various AmfTypes implemented here. It has two discrete
+// responsibilities:
+//	- Fetch the AmfType assosciated with a given marker ID.
+//	```
+//	typ := DefaultIdentifier.TypeOf(0x01)
+//	  #=> (reflect.TypeOf(new(amf0.Bool)).Elem())
+//	```
+//
+//	- Fetch the AmfType assosciated with a given native type. (i.e., going
+//	from a `bool` to an `amf0.Bool`.)
+//	```
+//	typ := DefaultIdentifier.AmfType(true)
+//	  #=> (reflect.TypeOf(new(amf0.Bool)).Elem())
+//	```
+type Identifier struct {
+	ids  map[byte]reflect.Type
+	typs map[reflect.Type]reflect.Type
```

Comment:
I thought about this while implementing. Unfortunately, `reflect.Kind` doesn't support types not in the stdlib, like `*amf0.Object`, if we wanted to use that (we'd just `reflect.Kind(Pointer)`, which is useless).